CARET_trace/src/ros_trace_points.cpp

```diff
@@ -51,6 +52,89 @@
 std::unique_ptr<std::thread> trace_node_thread;
 thread_local bool trace_filter_is_rcl_publish_recorded;
 
+using std::string;
+using std::vector;
+bool ignore_rcl_timer_init = false;
+
+vector<string> string_split(string & str, char delim)
+{
+  using std::stringstream;
+  vector<string> elems;
+  stringstream ss(str);
+  string item;
+  while (getline(ss, item, delim)) {
+    if (!item.empty()) {
+      elems.push_back(item);
+    }
+  }
+  return elems;
+}
+
+bool is_ros2_launch_command()
```

Comment:
It is better to add "static" to avoid conflict of global names.

```suggestion
static bool is_ros2_launch_command()
```

Reply:
i fixed